File: backup-workspace-20260203-1136/skills/weather/get-weather.ts
#!/usr/bin/env bun
/**
 * Weather fetcher using wttr.in API
 * Usage: pnpm tsx get-weather.ts [city]
 * Default city: Rennes
 */

const city = process.argv[2] || "Rennes";

interface WttrData {
  current_condition: Array<{
    temp_C: string;
    FeelsLikeC: string;
    humidity: string;
    windspeedKmph: string;
    weatherDesc: Array<{ value: string }>;
    uvIndex: string;
  }>;
  weather: Array<{
    date: string;
    maxtempC: string;
    mintempC: string;
    hourly: Array<{
      time: string;
      tempC: string;
      chanceofrain: string;
      weatherDesc: Array<{ value: string }>;
    }>;
  }>;
}

async function getWeather(city: string): Promise<string> {
  try {
    const url = `https://wttr.in/${encodeURIComponent(city)}?format=j1`;
    const response = await fetch(url, {
      headers: { "Accept-Language": "fr" }
    });
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    
    const data: WttrData = await response.json();
    const current = data.current_condition[0];
    const today = data.weather[0];
    
    // Current conditions
    const temp = current.temp_C;
    const feelsLike = current.FeelsLikeC;
    const humidity = current.humidity;
    const wind = current.windspeedKmph;
    const desc = current.weatherDesc[0]?.value || "?";
    const uv = current.uvIndex;
    
    // Today's forecast
    const maxTemp = today.maxtempC;
    const minTemp = today.mintempC;
    
    // Rain probability from hourly data
    const nextHours = today.hourly.slice(0, 4);
    const maxRainChance = Math.max(...nextHours.map(h => parseInt(h.chanceofrain) || 0));
    
    // Weather emoji based on description
    const descLower = desc.toLowerCase();
    let emoji = "🌤️";
    if (descLower.includes("rain") || descLower.includes("pluie")) emoji = "🌧️";
    else if (descLower.includes("cloud") || descLower.includes("nuag")) emoji = "⛅";
    else if (descLower.includes("sun") || descLower.includes("soleil") || descLower.includes("clear")) emoji = "☀️";
    else if (descLower.includes("snow") || descLower.includes("neige")) emoji = "❄️";
    else if (descLower.includes("thunder") || descLower.includes("orage")) emoji = "⛈️";
    else if (descLower.includes("fog") || descLower.includes("brouillard")) emoji = "🌫️";
    
    let output = `${emoji} ${desc} ${temp}°C`;
    
    if (parseInt(feelsLike) !== parseInt(temp)) {
      output += ` (ressenti ${feelsLike}°C)`;
    }
    
    output += `\n💧 Humidité ${humidity}% | 💨 Vent ${wind} km/h`;
    
    if (parseInt(uv) >= 5) {
      output += ` | ☀️ UV ${uv}`;
    }
    
    output += `\n📊 Min ${minTemp}°C → Max ${maxTemp}°C`;
    
    if (maxRainChance > 30) {
      output += `\n🌧️ Risque pluie: ${maxRainChance}%`;
    }
    
    return output;
  } catch (error: any) {
    return `[Météo indisponible: ${error.message}]`;
  }
}

getWeather(city).then(console.log);

File: temp/get_weather.py
import urllib.request
import json

cities = ['Rennes', 'Paris', 'Lille']

for city in cities:
    with urllib.request.urlopen(f'https://wttr.in/{city}?format=j1') as r:
        data = json.loads(r.read())
    
    curr = data['current_condition'][0]
    temp = curr['temp_C']
    desc = curr['weatherDesc'][0]['value']
    
    # Check for rain today
    today = data['weather'][0]
    rain_chance = 0
    for h in today.get('hourly', []):
        rain_chance = max(rain_chance, int(h.get('chanceofrain', 0)))
    
    rain_info = f"🌧️ {rain_chance}%" if rain_chance > 30 else "☀️"
    
    print(f"=== {city.upper()} ===")
    print(f"Actuel: {temp}°C {desc} {rain_info}")
    
    # 5 day forecast
    forecasts = []
    for day in data['weather'][1:6]:  # skip today, get next 5
        date = day['date']
        d = date.split('-')
        day_str = f"{d[2]}/{d[1]}"
        min_t = day.get('mintempC', '?')
        max_t = day.get('maxtempC', '?')
        # Check rain
        day_rain = 0
        for h in day.get('hourly', []):
            day_rain = max(day_rain, int(h.get('chanceofrain', 0)))
        rain_icon = "🌧" if day_rain > 50 else ""
        forecasts.append(f"{day_str}:{min_t}/{max_t}{rain_icon}")
    
    print("Prévisions: " + " | ".join(forecasts))
    print()
